### ISSUE TITLE
Delete temporary files and directories

### ISSUE THREAD

**Issue by andrrizzi** [choderalab/yank#311]
In many places in the code we use mkdtemp() and sometimes NamedTemporaryFile(delete=False) from the tempfile module without deleting them when we're done, but the [documentation](https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp) says the user is responsible for this.

I'm not sure if we're assuming that the operative system cleans up automatically the temp directory, but we should probably take care of this. Maybe an elegant way to implement it would be using a context manager in yank.utils?

``` python
@contextlib.contextmanager
def temporary_directory():
    tmp_dir = tempfile.mkdtemp()
    try:
        yield tmp_dir
    finally:
        shutil.rmtree(tmp_dir)

with temporary_directory as tmp_dir:
    do_stuff()
```

**Comment by jchodera** [comment]
This is a great idea!  We should definitely be cleaning up files.

**Comment by rmcgibbo** [comment]
Mdtraj has a context manager for this that also `cd`s into the directory,
which is useful if you're going to run a shell script or program that might
write output to the current directory.

**Comment by Lnaden** [comment]
I think this has been fixed by the MDTraj tmp dirs. @andrrizzi can you confirm? If so, please close. 

Thanks!

**Comment by andrrizzi** [comment]
We'll still need to check this! We are using the `mdtraj` function and the `openmoltools.utils` functions in few places, but not in others. Although, this may be limited to SystemBuilder.

**Comment by Lnaden** [comment]
There are a couple instances of `mkdtemp` and `NamedTemproraryFile` in the systembuilder file (which is being depreciated and repurposed), then there are a a few of those calls in the `test_X` files. Do we want the tests to delete the temporary files? Do we want to fix this issue before 1.0?

**Comment by andrrizzi** [comment]
> Do we want to fix this issue before 1.0?

Probably not. It's not critical since usually the `tmp/` folder gets cleaned up at regular times by the OS.

**Comment by Lnaden** [comment]
I think after all the refactors, this has been resolved. We can reopen if its a problem or I missed some
